internal/eventstore/eventstore.go

```diff
@@ -262,6 +268,12 @@ type Pusher interface {
 	Push(ctx context.Context, commands ...Command) (_ []Event, err error)
 }
 
+type Searcher interface {
+	// Search predefined results
+	// the instance is used from ctx
+	Search(ctx context.Context, conditions ...map[SearchFieldType]any) (result []*SearchResult, err error)
```

Comment:
So let me get this right:

The proposed interface only gives 1 or more search results. If we want the related events, we need to do another query using the aggregate?
So we will not have an interface that solves the following problem in a single databae call:

> I want event types `x`, `y` and `z` of the project aggregate where the clientID is `123`.

When we were having our beer talks I kind of had the above in mind. Eg, we do a search and then join all related events and return them.
We can always implement that in a later stage off course, after the current concept is proven.

Reply:
Yes, to be evaluated later. ATM this implementation solves the issues we have.